sql/planbuilder/scalar.go

```diff
@@ -169,6 +169,11 @@ func (b *Builder) buildScalar(inScope *scope, e ast.Expr) (ex sql.Expression) {
 			b.handleErr(err)
 		}
 
+		switch rf.(type) {
+		case *function.Sleep, sql.NonDeterministicExpression:
+			b.qFlags.Unset(sql.QFlagDeferProjections)
```

Comment:
it might be mildly safer to add a new flag, for the presence of volatile expressions